index.js

```diff
@@ -7,14 +10,287 @@ function ask(questionText) {
+      Plz, try again.`)
+  } 
+  // if action is known, let user know as well
+  else {
+    console.log(`Ah, so you want want to ${action} ${target}.`)
+
+    // if look - assume "look around" - get current place's location & look out! 
+    if ( action === 'look' ) {
+      // TODO explore using text wrapping - currently using `` and formatting as desired.
+      // https://stackoverflow.com/questions/14484787/wrap-text-in-javascript
+      console.log(locationLookUp[player.currentLocation].lookAround())
+    }
+    // ===     logic to move b/t locations ===== //
+    // if "go", expect/check target is location, 
+    else if ( action === "go to" || action === "go" ) {
+      // check/handle if valid location
+      if(Object.keys(locationLookUp).includes(target)) {
+        // check/handle if allowed transition
+        // TODO story -- when leaving main entrance to cart room, main entrance locks!
+        //               can't leave w/o the full shopping leave.  
+        // TODO story -- when shopper returns to main entrance w/ everything, unlock it! 
+        if (locationLookUp[player.currentLocation].canGo(target)) {
+          console.log(`Good guess! You left ${player.currentLocation} and are now in ${target}`)
+          player.currentLocation = target
+        } else {
+          // TODO implement can't go from here to there -- if we got a valid location
+          console.log(`Can't go from ${player.currentLocation} to ${target}. Clues, shopper, clues.`)
+        }
+      } else {
+        // wrong location -- help user with location
+        console.log(`Hmmm, don't know ${target}. Look around!`)
+      }   
+    }
+    // TODO #3 implement "take" actions, expect take [item]
+    else if ( action == "take" ) {
+      // non-produce items  -- shopping list, cart, inventory
+      if ( target === "shopping list") {
+        console.log(`Awesome! Here's what you need to get: `, shoppingList)
+      } else if ( target == "cart" ) {
+        console.log(`You got yourself a cart -- now let's get shopping!`)
+      } 
+      // TODO #3b add take inventory (as item) => list what user has
+      else if ( target == "inventory" ) {
+        console.log(`Here's what you got so far:`, player.cart)
+      } else {
+
+      }
+      
+
+      // taking a produce item
+
+    }
+    // TODO #4 if "drop" expect target = [item], check
+    else if ( action === 'drop') {
+
+    }
+    // TODO #5 if "pay", say thanks - you're all set
+    else if ( action === 'pay') console.log(
+      `
+      Thanks for the cashless exchange! You're all set. 
+      You can leave now. You remember where you came from, don't you?`)
+    // if "leave" say bye
+    else if (action === 'leave') { 
+      // TODO - Have they opened the main entrance? 
+      //        If not, give them a hard time and release them.
+
+      console.log("bye")
+      process.exit(0)
+    }
+
+  }
+
+  start()
+  // process.exit(); -- TODO understand how/when this line executes
 }
```

Comment:
You call the start function here again, it already hits on line 183

Reply:
Thanks for the feedback! I will return to check all this out.